### PR TITLE
feat(core): update sie

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

The final part of new update SIE api.

## Testing

UTs

### CODE REVIEW

File: packages/core/src/routes/sign-in-experience.ts [hunk 1]
```diff
@@ -67,6 +68,15 @@ export default function signInExperiencesRoutes<T extends AuthedRouter>(router:
         validateSignUp(signUp, enabledConnectors);
       }
 
+      if (signIn) {
+        if (signUp) {
+          validateSignIn(signIn, signUp, enabledConnectors);
+        } else {
+          const signInExperience = await findDefaultSignInExperience();
+          validateSignIn(signIn, signInExperience.signUp, enabledConnectors);
```
Comment (charIeszhao) [inline packages/core/src/routes/sign-in-experience.ts:76]: nit: maybe this?
```ts
const signUp = signUp ?? ((await findDefaultSignInExperience()).signUp);
```

**Review by charIeszhao — APPROVED** [review]

**Review by darcyYe — APPROVED** [review]

File: packages/core/src/routes/sign-in-experience.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/routes/sign-in-experience.ts:76]: if (signIn && signUp)

else if (signIn)

Comment (wangsijie) [reply packages/core/src/routes/sign-in-experience.ts:76]: @charIeszhao it's not allowd to do that.

Comment (wangsijie) [reply packages/core/src/routes/sign-in-experience.ts]: updated @simeng-li

**Pull request merged**